src/components/combo_box/combo_box_options_list/combo_box_options_list.tsx

```diff
@@ -232,15 +233,27 @@ export class EuiComboBoxOptionsList<T> extends Component<
           );
         }
       } else {
-        emptyStateContent = (
-          <p>
-            <EuiI18n
-              token="euiComboBoxOptionsList.noMatchingOptions"
-              default="{searchValue} doesn't match any options"
-              values={{ searchValue: <strong>{searchValue}</strong> }}
-            />
-          </p>
-        );
+        if (delimiter && searchValue.includes(delimiter)) {
+          emptyStateContent = (
+            <p>
+              <EuiI18n
+                token="euiComboBoxOptionsList.noMatchingOptions"
```

Comment:
You'll need to change the name of the token here too since you can't use the same one multiple times.
```suggestion
                token="euiComboBoxOptionsList.delimiterMessage"
```